wpilibj/src/main/java/edu/wpi/first/wpilibj/RobotBase.java

```diff
@@ -38,7 +38,7 @@
    * The ID of the main Java thread.
    */
   // This is usually 1, but it is best to make sure
-  public static final long MAIN_THREAD_ID = Thread.currentThread().getId();
+  private static long m_threadId = -1;
```

Comment:
I think it can still be final?  Since it's only set from the constructor...

Reply:
Its set from the instance constructor, not the static constructor. So it can't be final.